packages/exporter-trace-otlp-http/src/platform/node/util.ts

```diff
@@ -62,24 +74,47 @@ export function sendWithHttp<ExportItem, ServiceRequest>(
   const req = request(options, (res: http.IncomingMessage) => {
     let responseData = '';
     res.on('data', chunk => (responseData += chunk));
-    res.on('end', () => {
-      if (res.statusCode && res.statusCode < 299) {
-        diag.debug(`statusCode: ${res.statusCode}`, responseData);
-        onSuccess();
-      } else {
-        const error = new otlpTypes.OTLPExporterError(
-          res.statusMessage,
-          res.statusCode,
-          responseData
+
+    res.on('aborted', () => {
+      if (reqIsDestroyed) {
+        const err = new otlpTypes.OTLPExporterError(
+          'Request Timeout'
         );
-        onError(error);
+        onError(err);
       }
     });
-  });
 
+    res.on('end', () => {
+      // node version <= 12 will emit 'end' event even if request was aborted
+      // no need to handle promise here if request was aborted since promise
+      // was already rejected in res.on('aborted')
```

Comment:
The comment is not clear to me. It mentions a version diff for <= 12 but it seems there is no if checking the node version.